packages/components/src/tree-select/types.ts

```diff
@@ -18,7 +18,7 @@ export type SelectOptions = Required<
 >[ 'options' ];
 
 export interface Tree {
-	id: string;
+	id: number | string;
```

Comment:
This is actually passing to `TreeSelect` an `id` of type `number`:

<img width="966" alt="Screen Shot 2022-12-21 at 4 29 52 PM" src="https://user-images.githubusercontent.com/4063887/209022654-21b20518-6270-45e1-8932-e928b578bbe5.png">

Reply:
Otherwise, I could fix this typing issue in [packages/components/src/query-controls/](https://github.com/WordPress/gutenberg/tree/921135f9ec7f1e54030672eb7d11ca28034bf15b/packages/components/src/query-controls)

But that'll be more much more complex.

Reply:
Sure, how does [this conversion to string](https://github.com/WordPress/gutenberg/pull/46721/commits/2442dd543021b311896131673b755043ff9c0456#diff-c56c83975cc875f46b914669b834ebdfb277208a4d4a31f32ca1a6c51b709b91R24) look?

Reply:
Seems fine! No need to check against `undefined` here, since `term.id` _should_ be always defined according to the types.